src/modm/container/dynamic_array_impl.hpp

```diff
@@ -61,35 +61,35 @@ modm::DynamicArray<T, Allocator>::DynamicArray(const DynamicArray& other) :
 {
 	this->values = allocator.allocate(other.capacity);
 	for (SizeType i = 0; i < this->size; ++i) {
-		this->allocator.construct(&this->values[i], other.values[i]);
+		std::construct_at(&this->values[i], other.values[i]);
 	}
 }
 
 template <typename T, typename Allocator>
 modm::DynamicArray<T, Allocator>::~DynamicArray()
 {
 	for (SizeType i = 0; i < this->size; ++i) {
-		this->allocator.destroy(&this->values[i]);
+		std::destroy_at(&this->values[i]);
 	}
-	this->allocator.deallocate(this->values);
+	this->allocator.deallocate(this->values, 1);
```

Comment:
According to the [documentation](https://en.cppreference.com/w/cpp/memory/allocator/deallocate) deallocate must be called with the same count as the corresponding allocation, "otherwise, the behavior is undefined".
```suggestion
	this->allocator.deallocate(this->values, capacity);
```